pytorch_lightning/plugins/io/torch_plugin.py

```diff
@@ -30,6 +31,23 @@ class TorchCheckpointIO(CheckpointIO):
     respectively, common for most use cases."""
 
     def save_checkpoint(self, checkpoint: Dict[str, Any], path: _PATH, storage_options: Optional[Any] = None) -> None:
+        """Save model/training states as a checkpoint file through state-dump and file-write.
+
+        Args:
+            checkpoint: dict containing model and trainer state
+            path: write-target path
+            storage_options: not used in ``TorchCheckpointIO.save_checkpoint``
+
+        Raises:
+            MisconfigurationException:
+                If ``storage_options`` arg is passed in
+        """
+        if storage_options is not None:
+            raise MisconfigurationException(
```

Comment:
#10918 

```suggestion
            raise TypeError(
```

Same comment applies to the other usages